examples/mugen/retrieval/configs/eval.yaml

```diff
@@ -0,0 +1,39 @@
+dataset_args:
```

Comment:
Explore using structure config to manage the arguments: https://omegaconf.readthedocs.io/en/2.1_branch/structured_config.html#id1

At a first look, this will give you the benefits of:
- hierarchical data structures
- type safety
- handling of optional arguments

To get a sense of its usage:
- https://github.com/facebookresearch/multimodal/blob/2352bc5b63d30e7324a8547d2ee58eac4d48b163/examples/flava/definitions.py
- https://github.com/facebookresearch/multimodal/blob/main/examples/flava/configs/pretraining/debug.yaml

Reply:
the flava example follows the [hydra specs](https://hydra.cc/docs/advanced/instantiate_objects/overview/), so i updated the PR according to that